File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int mini=prices[0];
        int maxpro=0;
        for(int i=1;i<n;i++){
            int cost=prices[i]-mini;
            maxpro=Math.max(cost,maxpro);
            mini=Math.min(mini,prices[i]);
        }
        return maxpro;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        ans = prices[0]
        profit = 0
        for i in range(1, len(prices)):
            ans = min(ans, prices[i])
            profit = max(profit, prices[i] - ans)
        
        return profit

        